addon-a11y not working on latest next

**Describe the bug**
`@storybook/addon-a11y` not reporting a11y test results on `next` branch . It shows `0 passes` and `0 violations` on any story regardless of decorator being set or not.

**To Reproduce**
1. Checkout `next` branch
2. Start the `official-storybook` kitchen sink app or install/register add-on to any other app.
2. Open any of the `a11y` stories with decorator set.
3. Go to `Accessibility` panel.
4. See 0 results on both passes/violations.

**Expected behavior**
Should display amount of test passes or violations respectively in each tab.

**Screenshots**
![screen shot 2018-11-29 at 10 19 17 am](https://user-images.githubusercontent.com/1105953/49231781-a9a17000-f3c0-11e8-9451-4603a99f0912.png)

**Code**
It's related to the `runA11yCheck` function on the addon's declaration and the DOM being selected to run the aXe tests on:

```js
const runA11yCheck = () => {
  const channel = addons.getChannel();
  const infoWrapper = document.getElementById('story-root').children;
  const wrapper = document.getElementById('root');

  axe.reset();
  axe.configure(axeOptions);
  axe
    .run(infoWrapper || wrapper)
    .then(results => channel.emit(CHECK_EVENT_ID, results), logger.error);
};
```

**System:**
 - Browser: any
 - Framework: all
 - Addons: a11y
 - Version: 4.0.9

**Additional context**
Tried this on some of the other kitchen sink apps with same result.

This is also an issue on the [official Storybook example page](https://storybooks-official.netlify.com/?selectedKind=Addons%7Ca11y&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=%40storybook%2Faddon-a11y%2Fpanel)

I'm guessing this is related to #4704 - Adding a `<div id="root-story">` around all my stories, causes aXe to throw a selector warning, but it does run all the tests correctly afterwards. It still scrolls to the bottom of the page.

I'm not sure why the `root-story` is only added to the Info addon? Scrolling to the bottom of a story is a general issue, that's not just related to pages with the Info addon. The **readme** addon has the same issue, and in my project i have multiple stories that either render entire pages or components that just take up more than the screen.

```
index.js:19 Error: Expect axe._selectorData to be set up
    at generateSelector (axe.js:3953)
    at Object.createUniqueSelector [as getSelector] (axe.js:4009)
    at DqElement.get selector [as selector] (axe.js:3400)
    at axe.js:2393
    at Array.map (<anonymous>)
    at axe.js:2386
    at Array.map (<anonymous>)
    at axe.js:2383
    at Array.forEach (<anonymous>)
    at Object.push../node_modules/axe-core/axe.js.helpers.processAggregate (axe.js:2375)
```

--- a/addons/a11y/README.md
+++ b/addons/a11y/README.md
@@ -50,11 +50,13 @@ import { storiesOf } from '@storybook/react';
 
 import { checkA11y, configureA11y } from '@storybook/addon-a11y';
 
-const whateverOptionsYouWant = {
-  axeOptions : {},
-  contextElementId: ''
-};
-configureA11y(whateverOptionsYouWant);
+// axe options
+const whateverOptionsYouWant = {};
+
+// optional element id for axe context 
+const contextElementId = '';
+
+configureA11y(whateverOptionsYouWant, contextElementId);
 
 storiesOf('button', module)
   .addDecorator(checkA11y)

--- a/addons/a11y/README.md
+++ b/addons/a11y/README.md
@@ -50,7 +50,10 @@ import { storiesOf } from '@storybook/react';
 
 import { checkA11y, configureA11y } from '@storybook/addon-a11y';
 
-const whateverOptionsYouWant = {};
+const whateverOptionsYouWant = {
+  axeOptions : {},
+  contextElementId: ''
+};
 configureA11y(whateverOptionsYouWant);
 
 storiesOf('button', module)

--- a/addons/a11y/src/index.js
+++ b/addons/a11y/src/index.js
@@ -6,22 +6,28 @@ import { logger } from '@storybook/client-logger';
 
 import { CHECK_EVENT_ID, REQUEST_CHECK_EVENT_ID } from './shared';
 
-let axeOptions = {};
+const config = {
+  axeOptions: {},
+  contextElementId: '',
+};
 
 export const configureA11y = (options = {}) => {
-  axeOptions = options;
+  Object.assign(config, options);
 };
 
 const runA11yCheck = () => {
   const channel = addons.getChannel();
-  const infoWrapper = document.getElementById('story-root').children;
-  const wrapper = document.getElementById('root');
+
+  let axeContext = config.contextElementId;
+  if (!axeContext) {
+    const infoWrapper = document.getElementById('story-root');
+    const wrapper = document.getElementById('root');
+    axeContext = (infoWrapper && infoWrapper.children) || wrapper;
+  }
 
   axe.reset();
-  axe.configure(axeOptions);
-  axe
-    .run(infoWrapper || wrapper)
-    .then(results => channel.emit(CHECK_EVENT_ID, results), logger.error);
+  axe.configure(config.axeOptions);
+  axe.run(axeContext).then(results => channel.emit(CHECK_EVENT_ID, results), logger.error);
 };
 
 const a11ySubscription = () => {

--- a/addons/a11y/src/index.js
+++ b/addons/a11y/src/index.js
@@ -11,8 +11,9 @@ const config = {
   contextElementId: '',
 };
 
-export const configureA11y = (options = {}) => {
-  Object.assign(config, options);
+export const configureA11y = (axeOptions = {}, contextElementId = '') => {
+  config.axeOptions = axeOptions;
+  config.contextElementId = contextElementId;
 };
 
 const runA11yCheck = () => {

--- a/addons/a11y/src/index.js
+++ b/addons/a11y/src/index.js
@@ -8,23 +8,25 @@ import { CHECK_EVENT_ID, REQUEST_CHECK_EVENT_ID } from './shared';
 
 const config = {
   axeOptions: {},
-  contextElementId: '',
+  axeContext: '',
 };
 
-export const configureA11y = (axeOptions = {}, contextElementId = '') => {
+export const configureA11y = (axeOptions = {}, axeContext = '') => {
   config.axeOptions = axeOptions;
-  config.contextElementId = contextElementId;
+  config.axeContext = axeContext;
+};
+
+const getDefaultAxeContext = () => {
+  const infoWrapper = document.getElementById('story-root');
+  const wrapper = document.getElementById('root');
+
+  return (infoWrapper && infoWrapper.children) || wrapper;
 };
 
 const runA11yCheck = () => {
   const channel = addons.getChannel();
 
-  let axeContext = config.contextElementId;
-  if (!axeContext) {
-    const infoWrapper = document.getElementById('story-root');
-    const wrapper = document.getElementById('root');
-    axeContext = (infoWrapper && infoWrapper.children) || wrapper;
-  }
+  const axeContext = config.axeContext || getDefaultAxeContext();
 
   axe.reset();
   axe.configure(config.axeOptions);

